iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -214,6 +217,8 @@ inline bool vector<T, Capacity>::pop_back() noexcept
 {
     if (m_size > 0U)
     {
+        // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+        // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
         reinterpret_cast<const T*>(m_data)[--m_size].~T();
```

Comment:
Again this seems somehow wrong ... Why does a const cast work here? You cannot call the destructor on a const object since it changes the object itself.
Furthermore, since it is casted to a pointer, shouldn't the line be:
```cxx
reinterpret_cast<T*>(m_data)[--m_size]->~T();
```

Reply:
@elfenpiff this must be possible else RAII would not work with `const`
```cpp
{
    const Foo foo;
   // ... do something with foo
} // dtor of foo must be called
```

Reply:
@elfenpiff hehe
![grafik](https://user-images.githubusercontent.com/56729607/178570787-2c6e8860-1bf9-4327-a07a-1d91746ab914.png)

Reply:
@elfenpiff one is also not supposed to call the dtor explicitly. This is undefined behavior

```cpp
const Foo foo;
foo.~Foo();
foo.bar();
```

Similar to

```cpp
Foo foo1;
Foo foo2{std::move(foo1)};
foo1.bar();
```

Reply:
Used `at_unchecked` here.